### PR TITLE
Only start legacy provisioner on legacy CR

### PR DESCRIPTION
**Pull request by awels**
Signed-off-by: Alexander Wels <awels@redhat.com>

**What this PR does / why we need it**:
Also refactor some of the reconcile code to use common pattern and reduce code copy and paste.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Enhancement: Only start legacy provisioner if legacy CR is provided.
```

### CODE REVIEW

File: pkg/controller/hostpathprovisioner/controller.go [hunk 1]
```diff
@@ -277,7 +277,11 @@ func (r *ReconcileHostPathProvisioner) Reconcile(context context.Context, reques
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
-	reqLogger.Info("Reconciling CSI and legacy controller plugin")
+	if r.isLegacy(cr) {
+		reqLogger.Info("Detected legacy CR, Reconciling CSI and legacy controller plugin")
```
Comment (akalenyu) [inline pkg/controller/hostpathprovisioner/controller.go:281]: title says `Only start legacy provisioner on legacy CR` but here if you detect legacy,
will reconcile both CSI & legacy controller?

Comment (awels) [reply pkg/controller/hostpathprovisioner/controller.go:281]: Yes, the CSI driver is started always, but it only starts the legacy provisioner if the legacy CR is detected. This is mainly to facilitate upgrades.

**Pull request merged**